### DISCUSSION TITLE
binvar often needs to be declared in the beginning of the script

### DISCUSSION THREAD

**Discussion by zoharl3** [yalmip/YALMIP#987] [Possible bug or missing functionality] [answered]
Gurobi MILP reports infeasibility. I move
```
x = binvar(n, 1);
```
to the top of the script, and the infeasibility is resolved. It had happened quite often until I started the habit of declaring binvars early.

I don't have an example at the moment, but if it's interesting, I can try and come back in the future with something.

**Comment by johanlofberg** [comment, marked as answer]
That should not have any influence (except in some rare complicated mixed nonconvex quadratic + SOCP cones where there is a known bug causing an issue which I haven't fixed) so you would have to create a reproducible example
